website/addons/onedrive/tests/utils.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+import mock
+from contextlib import contextmanager
+
+from modularodm import storage
+
+from framework.mongo import set_up_storage
+
+from website.addons.base.testing import AddonTestCase
+from website.addons.onedrive import MODELS
+
+
+def init_storage():
+    set_up_storage(MODELS, storage_class=storage.MongoStorage)
+
+
+class OneDriveAddonTestCase(AddonTestCase):
+
+    ADDON_SHORT_NAME = 'onedrive'
+
```

Comment:
This class will need to be updated to use the generic tests -- it will need:
- to inherit from `OAuthAddonTestCaseMixin`
- `ExternalAccountFactory` and `Provider` defined,
- `set_user_settings` should probably go away in favor of using  `OAuthAddonTestCaseMixin.set_user_settings`, 
- `set_node_settings` will need a `super` call, and a `settings.save()` call.